### PR TITLE
feat: change button flow. Add validations and make it responsive

### PR DESCRIPTION
**Pull request by braianj**
Screenshot 2024-05-23 at 11 45 25 AM
Screenshot 2024-05-23 at 11 45 58 AM

### CODE REVIEW

File: src/modules/subscription/reducer.ts [hunk 1]
```diff
@@ -103,10 +103,13 @@ export function subscriptionReducer(state = buildInitialState(), action: Subscri
 
     case SAVE_SUBSCRIPTION_EMAIL_SUCCESS: {
       const { email } = action.payload
-
+      let unconfirmedEmail: string | undefined = email
```
Comment (LautaroPetaccio) [inline src/modules/subscription/reducer.ts:106]: Would you mind adding a test for this new case?

File: src/components/Notifications/NotificationEmailCard/NotificationEmailCard.tsx [hunk 2]
```diff
@@ -91,9 +94,15 @@ function NotificationEmailCard(props: Props) {
...
-                {hasConfirmEmail ? t('settings.notifications.email.confirmed') : t('settings.notifications.email.pending_approval')}
+            {unconfirmedEmail && !hasConfirmEmail && (
+              <SpanUnconfirmedEmail data-testid={NOTIFICATION_EMAIL_CARD_UNCONFIRMED_TEST_ID}>
+                {t('settings.notifications.email.pending_approval')}
+              </SpanUnconfirmedEmail>
+            )}
+
+            {hasConfirmEmail && !unconfirmedEmail && (
+              <SpanUnconfirmedEmail confirmed data-testid={NOTIFICATION_EMAIL_CARD_UNCONFIRMED_TEST_ID}>
+                {t('settings.notifications.email.confirmed')}
```
Comment (LautaroPetaccio) [inline src/components/Notifications/NotificationEmailCard/NotificationEmailCard.tsx:105]: As the component is the same, what if we add the conditionals to the text that needs to be rendered?

File: src/components/Notifications/NotificationEmailCard/NotificationEmailCard.tsx [hunk 3]
```diff
@@ -78,11 +81,11 @@ function NotificationEmailCard(props: Props) {
   const buttonText = useMemo(() => {
     if (!emailProp && !unconfirmedEmail && !hasConfirmEmail) {
       return t('settings.notifications.email.button_submit_label')
-    } else if (unconfirmedEmail) {
+    } else if (unconfirmedEmail && unconfirmedEmail === email) {
```
Comment (LautaroPetaccio) [inline src/components/Notifications/NotificationEmailCard/NotificationEmailCard.tsx:84]: Would you mind adding tests for the new conditions?

Comment (braianj) [reply src/components/Notifications/NotificationEmailCard/NotificationEmailCard.tsx:84]: We have that test. 

https://github.com/decentraland/account/blob/ebe4e22827b91ca397d708696b5701a4564c5cef/src/components/Notifications/NotificationEmailCard/NotificationEmailCard.spec.tsx#L125

**Review by marianogoldman — APPROVED** [review]

**Pull request merged**